### PR TITLE
Fix #1493: add align rule for Haskell

### LINKED ISSUE #1493: Provide an align command for the haskell-mode
Integrating something like the following would be a nice improvement (for those who don't want to bother with `hindent`)

The binding `<space>m a` could be used as it is the case in pupppet mode.

```
  (require 'align)
  ;; align rules
      (add-to-list 'align-rules-list
                   '(haskell-types
                     (regexp . "\\(\\s-+\\)\\(::\\|∷\\)\\s-+")
                     (modes quote (haskell-mode literate-haskell-mode))))
      (add-to-list 'align-rules-list
                   '(haskell-assignment
                     (regexp . "\\(\\s-+\\)=\\s-+")
                     (modes quote (haskell-mode literate-haskell-mode))))
      (add-to-list 'align-rules-list
                   '(haskell-arrows
                     (regexp . "\\(\\s-+\\)\\(->\\|→\\)\\s-+")
                     (modes quote (haskell-mode literate-haskell-mode))))
      (add-to-list 'align-rules-list
                   '(haskell-left-arrows
                     (regexp . "\\(\\s-+\\)\\(<-\\|←\\)\\s-+")
                     (modes quote (haskell-mode literate-haskell-mode))))
```

### PR DESCRIPTION
**Pull request by PierreR**

### CODE REVIEW

File: layers/+lang/haskell/packages.el [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 (setq haskell-packages
   '(
+    align
```
Comment (d12frosted) [inline layers/+lang/haskell/packages.el]: This has no effect until you define `haskell/init-align` function.

File: layers/+lang/haskell/packages.el [hunk 2]
```diff
@@ -189,7 +190,26 @@
     (add-hook 'evil-insert-state-entry-hook 'spacemacs//haskell-indentation-show-guides)
     (add-hook 'evil-emacs-state-entry-hook 'spacemacs//haskell-indentation-show-guides)
     (add-hook 'evil-insert-state-exit-hook 'spacemacs//haskell-indentation-hide-guides)
-    (add-hook 'evil-emacs-state-exit-hook 'spacemacs//haskell-indentation-hide-guides)))
+    (add-hook 'evil-emacs-state-exit-hook 'spacemacs//haskell-indentation-hide-guides))
+
+  ;; align rule for haskell
```
Comment (d12frosted) [inline layers/+lang/haskell/packages.el]: Also indentation is wrong here. :smile:

**Comment by d12frosted** [comment]
I would like to ask you to improve following things:

1. Remove `align` from list of packages, because it's not separate package and since you don't declare `haskell/init-align` it's redundant.
2. Fix indentation.

After that you can rebase your two commits into one and then force push into `align` branch - it will be automatically updated here.

Sorry if you are already familiar with git :smile:

**Comment by PierreR** [comment]
Are you sure the indentation is wrong ? It is on col2 together with `  (with-eval-after-load 'haskell-indentation ...`

**Comment by d12frosted** [comment]
Ah right. Sorry, my bad - something funky was going on with my mental parser :smile: Apologies.

**Comment by PierreR** [comment]
> Oh, also I am not sure about with-eval-after-load 'align. Is it necessary here? 
You seem to be right. I have added it because I have got a complain about unknown `align-rules-list` but now it seems to work fine without it ;-) So I will remove it.

**Comment by PierreR** [comment]
I have done `1`. Let me know if there is something else to change.

**Comment by d12frosted** [comment]
Thanks! And I don't have anything else. :smile:

File: layers/+lang/haskell/packages.el [hunk 3]
```diff
@@ -189,7 +189,26 @@
...
     (add-hook 'evil-insert-state-exit-hook 'spacemacs//haskell-indentation-hide-guides)
-    (add-hook 'evil-emacs-state-exit-hook 'spacemacs//haskell-indentation-hide-guides)))
+    (add-hook 'evil-emacs-state-exit-hook 'spacemacs//haskell-indentation-hide-guides))
+
+  ;; align rules for Haskell
+  (with-eval-after-load 'align
+    (add-to-list 'align-rules-list
+                 '(haskell-types
+                   (regexp . "\\(\\s-+\\)\\(::\\|∷\\)\\s-+")
+                   (modes quote (haskell-mode literate-haskell-mode))))
```
Comment (syl20bnr) [inline layers/+lang/haskell/packages.el]: Why `quote` ?
Can you replace them with `(modes . '(haskell-mode literate-haskell-mode))` ?

Comment (PierreR) [inline layers/+lang/haskell/packages.el]: Done. Thanks for the tips.

**Comment by syl20bnr** [comment]
Thank you :+1: 
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**